### DISCUSSION TITLE
Current state of Saleor discount functionality

### DISCUSSION THREAD

**Discussion by herobaby71** [saleor/saleor#12345] [General] [answered]
Hi Saleor team,

I am working on migrating the products from our old shop system to the Saleor platform. Many of our products (hundreds) are Pre-Ordered items with different sale end-date attached. However, I wasn't able to identify a way to perform an equivalent action on Saleor. 

Currently, Saleor has `Sale` and `Voucher`.  We will focus on `Sale` as it is most relevant to the above goal. While I can create a `Sale` with a custom end-date and attach a product to it, it is infeasible to create hundreds of custom `Sale` manually over the long term, especially when there is no convenient field on the Saleor Dashboard's product page.

**I have three questions related to the above functionality:**
1/ Do you guys know of a way to achieve the goal above using the existing functionality without adversely affecting the performance of the application?

2/ If so, is there an existing import plugin that can help us?

3/ It would also be good to know if such functionality is planned for the future.

Thank you for your time,

**Comment by timuric** [comment, marked as answer]
Hey @herobaby71 do you want to create orders that link to sale or just import orders with discount information? 
There is order import mutations in development right now that should be useful for you to create historical orders without writting directly to the database
https://github.com/saleor/saleor/issues/11870

**Reply by herobaby71** [reply]
For the `Order` import, attaching discount information should be fine since it has already been booked. I am more concerned with the `Product`, where each product is associated with a different `Sale`. Given that we update new products constantly with different Sale end dates, is the only way to create a new `Sale` object for each `Product`?

**Comment by michalmirumee** [comment]
Hi @herobaby71 if you would like to have a quick chat we could talk about your issues - we are a software agency working with Saleor. Would love to help :)
